src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -64,13 +67,73 @@ impl ops::AddAssign for ItemCount {
     }
 }
 
-#[derive(Default)]
 struct CoverageCalculator {
     items: BTreeMap<FileName, ItemCount>,
+    output_format: Option<OutputFormat>,
+}
+
+fn limit_filename_len(filename: String) -> String {
+    let nb_chars = filename.chars().count();
+    if nb_chars > 35 {
+        "...".to_string()
+            + &filename[filename.char_indices().nth(nb_chars - 32).map(|x| x.0).unwrap_or(0)..]
+    } else {
+        filename
+    }
+}
+
+#[derive(Serialize)]
+struct CoverageCalculatorEntry {
+    documented: u64,
+    total: u64,
+    percentage: f64,
```

Comment:
There's no reason to include the percentage in the JSON. Whatever is consuming the JSON can trivially calculate it if needed.

Reply:
This is a good point.